compute_endpoint/tests/unit/test_engines.py

```diff
@@ -361,3 +355,31 @@ def test_gcengine_rejects_resource_specification(task_uuid):
         ).result()
 
     assert "is not supported" in str(pyt_exc)
+
+
+def test_gcmpiengine_default_executor(randomstring):
+    with mock.patch.object(GlobusMPIEngine, "_ExecutorClass") as mock_ex:
+        mock_ex.__name__ = randomstring()
+        mock_ex.return_value = mock.Mock(launch_cmd="")
+        engine = GlobusMPIEngine()
+
+    assert mock_ex.called, "Expect Parsl's MPIExecutor as default option"
+
+    exp_en_label = GlobusMPIEngine.__name__
+    exp_ex_label = f"{exp_en_label}-{mock_ex.__name__}"
+    assert engine.label == exp_en_label, "Expect reasonable, explanatory default"
+    a, k = mock_ex.call_args
+    assert k["label"] == exp_ex_label, "Expect distinct executor label"
+    assert k["encrypted"] is True, "Expect encrypted by default"
+
+
+def test_gcmpiengine_accepts_resource_specification(task_uuid):
```

Comment:
This test just tests one more ``submit()`` called like test_gcmpiengine_default_executor right?

Why make it a separate test when the 4 test setup lines (4/6 -> majority) are shared?

It's reasonable to separate it I suppose but I'd add a check for ``task_uuid`` and ``"some": "conf"`` the resource spec itself as arguments

Reply:
> This test just tests one more submit() called like test_gcmpiengine_default_executor right?

No.  The `default_executor` is verifying `__init__` method logic.  This method is verifying `.submit` logic.  To the point, that's the whole raison d'être of _unit_ tests.  "I don't care _that_ it broke.  I care _where_ it broke."

There is, unfortunately, going to be some duplication between different unit tests.  Some of this can be pulled out into fixtures, but I've yet to discover how to deduplicate all related tests "beautifully."  If you discover that, well, educate me.